rental.rb

```diff
@@ -9,8 +9,64 @@ def initialize(date, book, person)
     person.rentals << self
   end
 
-  def self.all
-    ObjectSpace.each_object(self).to_a
-  end
   attr_accessor :date, :book, :person
+
+  def hash_book
+    book_hash = {
+      @book.title => {
+        'title' => @book.title,
+        'author' => @book.author,
+        'rentals' => @book.rentals
+      }
+    }
+    if @person.classroom
+      hash_student(book_hash)
+    else
+      hash_teacher(book_hash)
+    end
+  end
+
+  def hash_student(book_hash)
+    person_hash = {
+      @person.id => {
+        'age' => @person.age,
+        'name' => @person.name,
+        'p_p' => @person.parent_permission,
+        'id' => @person.id,
+        'classroom' => @person.classroom,
+        'rentals' => @person.rentals
+      }
+    }
+    tojson(book_hash, person_hash)
+  end
+
+  def hash_teacher(book_hash)
+    person_hash = {
+      @person.id => {
+        'age' => @person.age,
+        'name' => @person.name,
+        'specialization' => @person.specialization,
+        'id' => @person.id,
+        'rentals' => @person.rentals
+      }
+    }
+    tojson(book_hash, person_hash)
+  end
+
+  def tojson(book_hash, person_hash)
+    temp_hash = {
+      @date => {
+        'date' => @date,
+        'book' => book_hash,
+        'person' => person_hash
+      }
+    }
+    if File.zero?('rentals.json')
+      File.write('rentals.json', JSON.pretty_generate([temp_hash]))
+    else
+      data_from_json = JSON.parse(File.read('rentals.json'))
+      data_from_json = [data_from_json] if data_from_json.class != Array
+      File.write('rentals.json', JSON.pretty_generate(data_from_json << temp_hash))
+    end
```

Comment:
- [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20exit%20from%20your%20app%20save%20all%20the%20data%20(books%2C%20people%2C%20and%20rentals)%20into%20JSON%20files.), Please save the rentals data on exit only.

- [ ] There is an error that occurred when I try to add a new rental, So please fix that.

![image](https://user-images.githubusercontent.com/68796232/210468911-e8b80f5e-cad6-4eff-bf4d-e7afa8000926.png)